### ISSUE TITLE
wet canopy fraction is incorrectly a function of vegetation fraction

### ISSUE THREAD

**Issue by barlage** [NCAR/noahmp#91]
The partitioning of precipitation in precip_heat/CanopyWaterIntercept is scaled by fveg/VegFrac early in the module, but maxliq/CanopyLiqWaterMax (and similar for snow) are not scaled by fveg/VegFrac. This causes a dependence of fwet/CanopyWetFrac on vegetation fraction. I see two possible solutions:

1. the easy one is to scale capacity by vegetation fraction (I have tested this and it does solve the issue)
2. multiply by vegetation fraction later when water is input to the soil

I like solution 2 better since it is clearer to understand and the canopy water would be a physical value that is actually on the leaf and not spread across the grid.

**Comment by cenlinhe** [comment]
I have fixed this issue using the easy quick solution in this commit in release-v4.5-WRF branch: https://github.com/NCAR/noahmp/commit/0fd17e50a3cbdb4ddd6ba00196927803f656fed0
The fix in v5.0 (included in release-v5.0.1) is in this commit: https://github.com/NCAR/noahmp/commit/a7dd399972fca4d477c43654d384287ad62114a2

**Comment by chenghaow** [comment]
@barlage @cenlinhe I see a potential consistency issue in how `CanopyLiqWaterMax` and `InterceptCanopyRain` are calculated in `src/CanopyWaterInterceptMod.F90`. If the input LAI and SAI are already grid-averaged (which should be the case), then it makes sense to me that 

https://github.com/NCAR/noahmp/blob/70ffdc62df82b68234a8f6046a920eb1ea01d909/src/CanopyWaterInterceptMod.F90#L86

as `VegFrac` captures the LAI effect when `OptDynamicVeg` = 2/3/8: https://github.com/NCAR/noahmp/blob/70ffdc62df82b68234a8f6046a920eb1ea01d909/src/PhenologyMainMod.F90#L138

But I'm not sure about the following line
https://github.com/NCAR/noahmp/blob/70ffdc62df82b68234a8f6046a920eb1ea01d909/src/CanopyWaterInterceptMod.F90#L82
Given that `CanopyLiqHoldCap` is defined as "**maximum intercepted liquid water per unit veg area index [mm]**", 

https://github.com/NCAR/noahmp/blob/70ffdc62df82b68234a8f6046a920eb1ea01d909/src/CanopyWaterInterceptMod.F90#L43

if `LeafAreaIndEff` and `StemAreaIndEff` are already grid-averages, wouldn't multiplying by another `VegFrac` double count the LAI effect?

**Comment by cenlinhe** [comment]
For Noah-MP, the LAI and SAI used currently in the model should be the stand-scale LAI and SAI rather than grid-mean LAI and SAI. The LAI and SAI in the parameter table is the stand-scale value (or e.g., 100% veg covered grid LAI value from remote sensing).

**Comment by chenghaow** [comment]
Thanks Cenlin. In this case, the following calculation for `CanopyLiqWaterMax` makes sense to me, as `VegFrac` scales this up to grid-mean

https://github.com/NCAR/noahmp/blob/70ffdc62df82b68234a8f6046a920eb1ea01d909/src/CanopyWaterInterceptMod.F90#L82

But for `InterceptCanopyRain`

https://github.com/NCAR/noahmp/blob/70ffdc62df82b68234a8f6046a920eb1ea01d909/src/CanopyWaterInterceptMod.F90#L86

which essentially is
`InterceptCanopyRain = (1.0 - exp(-0.52 * (LeafAreaIndex + StemAreaIndex))) * RainfallRefHeight * PrecipAreaFrac`

Then it would only apply at the stand scale, not the grid-mean, because not all precipitation would be intercepted if LAI is low. Does that mean we need to multiply it by an additional vegetation fraction?

**Comment by cenlinhe** [comment]
Hi Chenghao, this is a good point. I am not sure how the original VegFrac parameterization (1.0 - exp(-0.52 * (LeafAreaIndex + StemAreaIndex)) was derived, in terms of whether it relates the stand-scale LAI to grid-level VegFrac or grid-level LAI to grid-level VegFrac. If it is the former, the current formulation should be fine. If it is the latter, we need to convert the stand-scale LAI in the dynamic vegetation growth scheme to grid-level LAI first inside dynamic vegetation growth scheme, and then use that VegFrac parameterization to compute grid-level VegFrac. 

@barlage @tslin2 @CharlesZheZhang What do you think?

**Comment by CharlesZheZhang** [comment]
Hi Cenlin and Chenghao, I can find the VegFrac parameterization from the NoahMP paper in 2011 (http://dx.doi.org/10.1029/2010JD015139) Eq. 11, also see Dickinson 1984 (https://agupubs-onlinelibrary-wiley-com.cuucar.idm.oclc.org/doi/pdf/10.1029/GM029) see below definition:

![Image](https://github.com/user-attachments/assets/a5b26aab-d4db-437c-a3e7-14c16282ff60)

While by definition, LAI is the surface area of transpiring vegetation per unit vegetated area, so it has a unit of m2/m2 or no unit. So it should be in stand-scale and we use this VegFrac to scale it to model grid. 
Hope this can help.

**Comment by cenlinhe** [comment]
Thank you, Zhe! Looks like in the current Noah-MP dynamic veg scheme, it follows the Dickinson 1984 definition and formulation, so the equation mentioned above by Chenghao is to connect stand-scale LAI to grid-level vegetation fraction (?). However, those empirical coefficients used in the formulation can be re-tuned or re-optimized. In fact, the whole dynamic veg scheme may need a comprehensive analysis for parameterization optimization before applying to specific cases.

**Comment by chenghaow** [comment]
Thanks, Zhe and Cenlin. I'd like to further clarify my earlier comment: 

For a **stand-scale** LAI + SAI, the maximum intercepted liquid water should be calculated as `CanopyLiqHoldCap * (LeafAreaIndEff + StemAreaIndEff)`. To scale this up to the **grid-level**, you need to multiply it by `VegFrac`. This is exactly what the following code does
https://github.com/NCAR/noahmp/blob/70ffdc62df82b68234a8f6046a920eb1ea01d909/src/CanopyWaterInterceptMod.F90#L82

Similarly, the **stand-scale** intercepted rainfall should be determined by both **stand-scale** LAI + SAI and precipitation rate, which should be something like `RainfallRefHeight * PrecipAreaFrac * Factor`, where `Factor` is an empirical extinction term that captures the exponential decay in how interception saturates with increasing LAI + SAI. Usually we use the form `(1.0 - exp(-k * (LAI + SAI)))`. This is similar to how `VegFrac` is calculated in Noah-MP. So essentially the following code
https://github.com/NCAR/noahmp/blob/70ffdc62df82b68234a8f6046a920eb1ea01d909/src/CanopyWaterInterceptMod.F90#L86

calculates the **stand-scale** intercepted rainfall rate, not the **grid-scale** rate. 

If my interpretation is correct, we likely need an additional scaling factor to convert this stand-scale InterceptCanopyRain to grid-scale...

**Comment by cenlinhe** [comment]
Hi Chenghao, now I understand your point. I agree with you on this. However, I think what is missing here in InterceptCanopyRain is not vegetation fraction but the empirical extinction factor you mentioned. For dynamic veg scheme, this  extinction factor happens to have a similar format as the vegetation fraction calculation. If you think about using satellite vegFrac map directly as input, then this InterceptCanopyRain already accounts for the vegFrac conversion from stand-scale to grid-scale. Here, the default equation essentially assumes 1.0 for the empirical extinction factor for maximum interception capability as noted in the comment. We may need to add the empirical extinction factor to this equation after a more comprehensive evaluation using observations to determine the best coefficient to be used in the empirical extinction factor formulation. Please let me know if my understanding makes sense to you.

**Comment by chenghaow** [comment]
Hi Cenlin, Yes, we don't need to introduce an additional coefficient if the input LAI/SAI values are already at the grid level, which I believe is the case based on Niu et al. (2011) ([https://doi.org/10.1029/2010JD015140](url)) - the GVF was designed to capture between-canopy gap. This is in fact consistent with early studies that used this empirical extinction function (e.g., van Dijk and Bruijnzeel, 2001). So as long as the LAI input is at the grid level, there's no need to adjust anything. 

The nuance arises when the LAI input is at the stand level. In that case, setting the empirical extinction factor to 1 essentially assumes that all precipitation falling on that vegetated portion is intercepted, regardless of the actual LAI + SAI value, as long as the canopy storage is <= CanopyLiqWaterMax. In other words, it assumes a fully closed canopy with no gaps between leaves. This will likely lead to overestimation of interception, especially when modeling vegetation in urban canopy models (something we've resolved in our version).

**Comment by cenlinhe** [comment]
Yes, what you said makes sense to me. We will come back to this issue for the non-urban Noah-MP treatment, when we have time to assess the model canopy interception process in more depth.

**Comment by cenlinhe** [comment]
We have used the first solution proposed by Mike above to fix this issue in an earlier PR as discussed above. We hope that future model improvements of using a subgrid Mosaic scheme would be a better general solution to separate vegetated vs. non-vegetated sub-grids. So I am closing this issue now. Regarding the canopy interception issue mentioned above, we will need to revisit this in future studies to systematically assess the formulation. Like mentioned above, current formulation assumes an empirical extinction factor of 1.0, which will likely lead to overestimation of interception.
